genie-core/src/main/java/com/netflix/genie/core/jpa/entities/JobRequestEntity.java

```diff
@@ -361,6 +366,44 @@ public void setMemory(@Min(1) final int memory) {
     }
 
     /**
+     * Get the applications to use for this job as a List of ids.
+     *
+     * @return The applications for the job
+     * @throws GenieException On any exception
+     */
+    public List<String> getApplicationsAsList() throws GenieException {
+        return JsonUtils.unmarshall(this.applications, LIST_STRING_TYPE_REFERENCE);
+    }
+
+    /**
+     * Sets the dependencies for the job request from a set of strings.
+     *
+     * @param applicationsList Application IDs for the job
+     * @throws GenieException for any processing error
+     */
+    public void setApplicationsFromList(final List<String> applicationsList) throws GenieException {
+        this.applications = applicationsList == null ? EMPTY_JSON_ARRAY : JsonUtils.marshall(applicationsList);
+    }
+
+    /**
+     * Gets the applications for the job as JSON array.
+     *
+     * @return applications
+     */
+    protected String getApplications() {
+        return this.applications;
+    }
+
+    /**
+     * Sets the dependencies for the job.
```

Comment:
Two different things. Dependencies are the files (queries, etc) the job needs to run. Applications are the id's of applications the user wants to override the default set of applications for the selected command.